consensus/propagation/have_wants.go

```diff
@@ -37,25 +36,28 @@ func (blockProp *Reactor) handleHaves(peer p2p.ID, haves *proptypes.HaveParts, b
 		blockProp.Logger.Error("peer not found", "peer", peer)
 		return
 	}
+
 	_, parts, fullReqs, has := blockProp.getAllState(height, round)
 	if !has {
 		// TODO disconnect from the peer
 		blockProp.Logger.Error("received part state for unknown proposal", "peer", peer, "height", height, "round", round)
 		return
 	}
 
-	blockProp.mtx.RLock()
-	defer blockProp.mtx.RUnlock()
+	p.Initialize(height, round, int(parts.Total()))
```

Comment:
what if the peer state is already initialized with a different size? this could happen if we receive a malicious have.

Reply:
good thinking, isn't this initialized using the size from the combined partset though? which is signed over from the proposer?

Reply:
makes sense, if we verify the signature then we should be good